src/main/java/com/google/sps/servlets/GetNearbyVendorsServlet.java

```diff
@@ -126,18 +143,14 @@ private Query buildGeoQuery(String prefixGeoHash, boolean hasDelivery) {
     return nearbyVendors;
   }
 
-  /** Computes the distance between two geographical points using Haversine formula */
-  private float computeDistance(GeoPt pointA, GeoPt pointB) {
-    double latitudeRadiansA = Math.toRadians(pointA.getLatitude());
-    double latitudeRadiansB = Math.toRadians(pointB.getLatitude());
-
-    double latitudeDifference = Math.toRadians(pointB.getLatitude() - pointA.getLatitude());
-    double longitudeDifference = Math.toRadians(pointB.getLongitude() - pointA.getLongitude());
-
-    double a = Math.pow(Math.sin(latitudeDifference / 2), 2) + 
-        Math.cos(latitudeRadiansA) * Math.cos(latitudeRadiansB) * Math.pow(Math.sin(longitudeDifference / 2), 2);
-    double c = 2 * Math.asin(Math.sqrt(a));
-
-    return (float) (GeoPt.EARTH_RADIUS_METERS * c);
+  /**
+   * Returns true if the time is between the salecard's opening hours
+   * E.g. if isOpenDuringNight, currentTime between (startTime, 23:59] or [00:00, endTime)
+   * else, currentTime between (startTime, endTime)
+   */
+  private boolean isBetweenOpeningHours(SaleCard saleCard, LocalTime currentTime) {
+    return (saleCard.isOpenDuringNight())
```

Comment:
Do we really need this variable when calculating if the business is open?

Reply:
We do not need to store it, but we do need to know if `startTime` is after `endTime` or not.
If not, we can easily know that `currentTime` needs to be in the range `(startTime, endTime)`
If yes, `currentTime` needs to be in the range `(startTime, 23:59)` OR `(currentTime, endTime)`.
This is because time's comparison are done based on a "line" with range `[00:00, 23:59]`.